content/automate/ManageIQ/Transformation/Infrastructure/VM/vmwarews.class/__methods__/utils.rb

```diff
@@ -22,14 +22,14 @@ def self.host_fingerprint(host)
                 Digest::SHA1.hexdigest(cert.to_der).upcase.scan(/../).join(":")
               end
 
-              def self.provider_connection(ems)
+              def self.provider_connection(ems, handle = $evm)
                 @provider_connection ||= RbVmomi::VIM.connect(:host => ems.ipaddress || ems.hostname, :user => ems.authentication_userid, :password => ems.authentication_password, :insecure => true)
               rescue => e
                 raise "Could not connect to #{ems.name}: #{e.message}" if c.nil?
               end
               private_class_method :provider_connection
 
-              def self.vm_get_ref(vim, vm)
+              def self.vm_get_ref(vim, vm, handle = $evm)
```

Comment:
@fdupont-redhat would you be logging these errors or just the raise an error. If you are going to be logging or using your shared log_and_raise the handle will get used. I am thinking these are all class methods

Reply:
@mkanoor These are indeed class methods, so I will log the errors from the catch all rescue in the calling method, eventually using shared log_and_raise.